### DISCUSSION TITLE
Bookmark zooms to pixel level

### DISCUSSION THREAD

**Discussion by meteorquake** [sumatrapdfreader/sumatrapdf#3893] [General]
I have an issue where when I click a bookmark on certain files (the bookmarks were added via Foxit Reader) Sumatra jumps to the page but goes to a gigantic zoom that shows the page at pixel level and you have to zoom out a long long way or click one of the auto zooms. So the bookmarks become rather unfriendly. They used to all work fine on Sumatra till a certain while back, now some do others don't. I attach an example -

[valeriana-4-560-DETAILS.pdf](https://github.com/sumatrapdfreader/sumatrapdf/files/13403468/valeriana-4-560-DETAILS.pdf)

**Comment by GitHubRulesOK** [comment]
hmm the bookmarks will zoom in any viewer but should all zoom to `149.27%` SumatraPDF is not respecting that and sees it as maximum zoom, unclear yet if thats from MuPDF or not

each typical entry looks valid enough as X Y Z so perhaps a multiplier problem
```
<</S/GoTo/D[2 0 R/XYZ 0 307.9 1.49268]/Type/Action>>
```

Elevating to an Issue see https://github.com/sumatrapdfreader/sumatrapdf/issues/3894

**Reply by GitHubRulesOK** [reply]
@meteorquake 
can you check tomorrows pre-release (about 2am GMT) should have fixed that specific zoom issue

**Reply by GitHubRulesOK** [reply]
**Bookmark zoom scale 100 twice bug is now closed.** 
The problem was Zoom times 1.49  was applied as 149 i.e 1.49 became scaled 100% twice = 14900%

**Comment by mikeyww** [comment]
In my settings file, when I use ZoomIncrement = 0 with custom ZoomLevels, the custom ZoomLevels are not always used. I zoom in a couple of times, and the Zoom then jumps directly to Fit Width instead of to my next custom level.

**Comment by mikeyww** [comment]
OK. Thank you for the reply and information. If I understand you correctly, you are indicating that a specific PDF file may override the zoom settings in the SumatraPDF configuration file.

**Reply by GitHubRulesOK** [reply]
@mikeyww 
It is possible there is an odd behaviour but if you have a file with different page ratios the behaviour may seem odder than if all are same ratio. since say you fit width then step in and out via toolbar it will be 3 stops not the expected 2 but with pages slightly offbeat and near their boundary (continuous) that could be 4 stops. 

The reported behaviour here is a bug that **on using a bookmark** the zoom is distorted to the extreme.

**Reply by mikeyww** [reply]
I am not fitting the width. I am just stepping through zoom levels. The program unexpectedly zooms to Fit Width, though I never specified that in my settings. I am not using a bookmark.

**Comment by mikeyww** [comment]
I will check it soon. Thank you very much.

Mike

**Reply by GitHubRulesOK** [reply]
@mikeyww 
the update is unlikely to sort your observation as said the fix was for the sample provided where bookmarks are not zooming to 149.268%.

**Comment by meteorquake** [comment]
Cheers for all that, I can confirm 3.6.15920 64bit doesn't pixel-zoom when choosing a bookmark on the file I've just tried, I checked it did on that particular file just before updating.
Can I mention that whilst solving that problem there seems to be a new bug, which is that Sumatra is now no longer opening maximised, which it used to do; it now fits the window to the page... I like to have all windows maximised unless there's a need to tile them.

**Reply by meteorquake** [reply]
Files are generally viewed by double-clicking a PDF in explorer or equivalent file manager, so there's no maximised option. They used to come out maximised before the update, but now they resize to part screen.
For example, I have an older version of Sumatra lurking on my drive (15899). If a drag a PDF onto it it opens maximised. If I do the same to the new one it won't, even if I have maximised and closed it or priorly launched it with a maximise shortcut and closed it.
So it seems to me something has changed :) It would be good to have it back to maximising, as I'm currently having to click to maximise every file I open.

**Reply by meteorquake** [reply]
When I launch WindowState is being ignored and on closing duly changed from 2 to 1.
I think to emphasise this point, if I delete the 4 files of sumatra 15920 then copy in the 4 from 15899, the PDFs correctly launch maximised (when double-clicked). If I then run Sumatra's update so it goes to 15920, the files change to launching part window. They are both using the same ini file as can be seen by the file that's opening using advanced settings.
I'm presuming this could be somehow related to the bug mentioned earlier: "What does seem to be the problem is command line start with -presentation works but not -fullscreen which starts as presentation !"
To get round it I would presumably need to make a wrapping program that takes an argument and launches sumatra maximised. Meanwhile I'll probably revert to the old version though that suffers from the pixel zoom. But I'll presume the cause of the problem will be spotted soon :)

**Reply by kjk** [reply]
hopefully fixed in https://github.com/sumatrapdfreader/sumatrapdf/commit/ea9d8deebe4c592a637288ac4793a8188d481c3c
